src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/ProfessionalUserInternalController.java

```diff
@@ -108,4 +113,43 @@
 
         return retrieveUserByEmail(email);
     }
+
+    @ApiOperation(
+            value = "Modify roles for user",
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            }
+    )
+    @ApiResponses({
+            @ApiResponse(
+                    code = 201,
+                    message = "User Roles has been added",
+                    response = OrganisationResponse.class
+            ),
+            @ApiResponse(
+                    code = 403,
+                    message = "Forbidden Error: Access denied"
+            ),
+            @ApiResponse(
+                    code = 404,
+                    message = "Not Found"
+            )
+    })
+    @PutMapping(
+            path = "/{orgId}/users/{userId}",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE
+    )
+    @Secured("prd-admin")
+    public ResponseEntity<ModifyUserRolesResponse> modifyRolesForExistingUserOfOrganisation(
+            @RequestBody ModifyUserProfileData modifyUserProfileData,
+            @PathVariable("orgId")  String orgId,
+            @PathVariable("userId") String userId
+    ) {
+
+        log.info("Received request to update user roles of an organisation...");
+        profExtUsrReqValidator.validateModifyRolesRequest(modifyUserProfileData, userId);

```

Comment:
Similar comments for external controller for BAD request swagger entry and validation